src/drivers/batt_smbus/batt_smbus.cpp

```diff
@@ -426,14 +677,42 @@ BATT_SMBUS::cycle()
 			}
 		}
 
+		// if it is a solo battery, check for shutdown on button press
+		if (_is_solo_battery) {
+			// read the button press indicator
+			if (read_block(BATT_SMBUS_MANUFACTURER_DATA, buff, 6, false) == 6) {
+				bool pressed = (buff[1] >> 3) & 0x01;
+
+				if (_button_press_counts >= ((BATT_SMBUS_BUTTON_DEBOUNCE_MS * 1000) / BATT_SMBUS_MEASUREMENT_INTERVAL_US)) {
+					// battery will power off
+					new_report.is_powering_off = true;
+
+					// warn only once
+					if (_button_press_counts++ == ((BATT_SMBUS_BUTTON_DEBOUNCE_MS * 1000) / BATT_SMBUS_MEASUREMENT_INTERVAL_US)) {
+						warnx("system is shutting down NOW...");
+					}
+
+				} else if (pressed) {
+					// battery will power off if the button is held
+					_button_press_counts++;
+
+				} else {
+					// button released early, reset counters
+					_button_press_counts = 0;
+					new_report.is_powering_off = false;
+				}
+			}
+		}
```

Comment:
This check for the button press is the same check that was performed on the 3drobotics/PX4Firmware-solo@8bd17a4. It looks at the `ManufacturerData` and determines if the battery button has been pressed.